src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -1291,7 +1293,10 @@ private void NotifyFeaturesAboutStateChange(DatabaseRecord record, long index)
 
                         SetUnusedDatabaseIds(record);
                         InitializeFromDatabaseRecord(record);
-                        IndexStore.HandleDatabaseRecordChange(record, index);
+
+                        if(handleIndexesChange)
+                            IndexStore.HandleDatabaseRecordChange(record, index);
```

Comment:
1. Can you elaborate more about this change? Why did it cause to create multiple replacement indexes?
2. I have some doubts regarding the name of the parameter `handleIndexesChange`. I wonder if we can have something more meaningful. Maybe it will become more clear after answering question 1.
3. Minor formatting issue - missing space after `if`

Reply:
This change was removed